packages/presentation/src/drawing.ts

```diff
@@ -0,0 +1,398 @@
+    let r = this.ctx.lineWidth / 2
+    if (!erasing) {
+      // Single point looks too small compared to a line of the same width
+      // So make it a bit biggers
+      r *= 1.5
+    }
+    this.ctx.lineWidth = 0
+    this.ctx.fillStyle = this.ctx.strokeStyle
+    this.ctx.arc(p.x, p.y, r, 0, Math.PI * 2)
+    this.ctx.fill()
+  }
+
+  drawSmoothSegment = (points: Point[], index: number, lastPoint: boolean): void => {
+    const curPos = points[index]
+    const prevPos = points[index - 1]
+    const avg = avgPoint(prevPos, curPos)
+    if (index === 1) {
+      this.ctx.moveTo(prevPos.x, prevPos.y)
+      if (lastPoint) {
+        this.ctx.lineTo(curPos.x, curPos.y)
+      } else {
+        this.ctx.quadraticCurveTo(curPos.x, curPos.y, avg.x, avg.y)
+      }
+    } else {
+      const prevAvg = avgPoint(points[index - 2], prevPos)
+      this.ctx.moveTo(prevAvg.x, prevAvg.y)
+      if (lastPoint) {
+        this.ctx.quadraticCurveTo(prevPos.x, prevPos.y, curPos.x, curPos.y)
+      } else {
+        this.ctx.quadraticCurveTo(prevPos.x, prevPos.y, avg.x, avg.y)
+      }
+    }
+  }
+}
+
+export function drawing (node: HTMLElement, props: DrawingProps): any {
+  if (
+    props.imageWidth === undefined ||
+    props.imageHeight === undefined ||
+    node.clientWidth === undefined ||
+    node.clientHeight === undefined
+  ) {
+    console.error('Failed to create drawing: image size is not specified')
+    return
+  }
+
+  const canvas = document.createElement('canvas')
+  canvas.style.position = 'absolute'
+  canvas.style.left = '0'
+  canvas.style.top = '0'
+  canvas.style.width = '100%'
+  canvas.style.height = '100%'
+  canvas.width = props.imageWidth
+  canvas.height = props.imageHeight
+  node.appendChild(canvas)
+
+  const ctx = canvas.getContext('2d')
+  if (ctx === null) {
+    console.error('Failed to create drawing: unable to get 2d canvas context')
+    node.removeChild(canvas)
+    return
+  }
+
+  const canvasCursor = document.createElement('div')
+  canvasCursor.style.visibility = 'hidden'
+  canvasCursor.style.position = 'absolute'
+  canvasCursor.style.borderRadius = '50%'
+  canvasCursor.style.cursor = 'none'
+  canvasCursor.style.pointerEvents = 'none'
+  node.appendChild(canvasCursor)
+
+  let readonly = props.readonly ?? false
+  let prevPos: Point = { x: 0, y: 0 }
+
+  const draw = new DrawState(ctx)
+  draw.tool = props.drawingTool ?? 'pan'
+  draw.penColor = props.penColor ?? 'blue'
+  updateCanvasCursor()
+
+  let modified = false
+  let commands: DrawCmd[] = []
+  let drawingData = props.drawingData
+  parseData()
+
+  const resizeObserver = new ResizeObserver((entries) => {
+    for (const entry of entries) {
+      if (entry.target === canvas) {
+        draw.scale = {
+          x: canvas.width / entry.contentRect.width,
+          y: canvas.height / entry.contentRect.height
+        }
+      }
+    }
+  })
+  resizeObserver.observe(canvas)
+
+  canvas.onpointerdown = (e) => {
```

Comment:
do we need to throttle this so we don't produce too many points for small changes?

Reply:
Yes, we have that, see `minLineLength`. 
This also helps to make lines smoother. A bigger value makes lines even more smoother, but in the same time gives an impression of  like 'slow' lines. So this an empirical value based on my taste :) could'be audjusted

Reply:
Maybe you already have this, but to reduce this "slow" lines we can use the current cursor position as the last point (no need to save it until `minLineLength` or mouse up).